src/UNL/Officefinder/Auth.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+class UNL_Officefinder_Auth
+{
+    const SESSION_MAP_CACHE_PREFIX = 'unl_cas_map_';
+
+    protected $cache;
+
+    public function __construct()
+    {
+        if (phpCAS::isInitialized()) {
+            return;
+        }
+
+        phpCAS::client(CAS_VERSION_2_0, 'login.unl.edu', 443, 'cas', false);
+        phpCAS::setPostAuthenticateCallback([$this, 'handleLoginTicket']);
+        phpCAS::setSingleSignoutCallback([$this, 'handleLogoutTicket']);
+        phpCAS::setCasServerCACert(GuzzleHttp\default_ca_bundle());
+    }
+
+    public function handleSingleLogOut()
+    {
+        phpCAS::handleLogoutRequests(false);
+    }
+
+    public function isAuthenticated()
+    {
+        return phpCAS::isAuthenticated();
+    }
+
+    public function getUsername()
+    {
+        return phpCAS::getUser();
+    }
+
+    public function gatewayAuthentication()
+    {
+        return phpCAS::checkAuthentication();
+    }
+
+    public function forceAuthentication()
+    {
+        return phpCAS::forceAuthentication();
+    }
+
+    public function logout($url = '')
```

Comment:
I probably should remove the param, just a b/c habit. The CAS server does not use it and CAS Protocol 3 says not to use it either.